cmd/maya-apiserver/cstor-operator/cspc/handler.go

```diff
@@ -160,3 +192,51 @@ func (pc *PoolConfig) create(pendingPoolCount int, cspc *apis.CStorPoolCluster)
 	}
 	return nil
 }
+
+// getUsedBDCs returns BDCList that is associated with the given cspc
+func (c *Controller) getUsedBDCs(cspc *apis.CStorPoolCluster, namespace string) (*ndmapis.BlockDeviceClaimList, error) {
+	bdcList, err := c.ndmclientset.OpenebsV1alpha1().BlockDeviceClaims(namespace).
+		List(metav1.ListOptions{LabelSelector: string(apis.CStorPoolClusterCPK) + "=" + cspc.Name})
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not list BDCs for CSPC %v", cspc.Name)
+	}
+	return bdcList, nil
+}
+
+// removeBDCFinalizer removes cspc finalizer from the BDCs
+func (c *Controller) removeBDCFinalizer(bdcObj *ndmapis.BlockDeviceClaim, finalizer string) (*ndmapis.BlockDeviceClaim, error) {
+	if len(bdcObj.Finalizers) == 0 {
+		return bdcObj, nil
+	}
+
+	bdcCopy := bdcObj.DeepCopy()
+	bdcCopy.Finalizers = util.RemoveString(bdcCopy.Finalizers, finalizer)
+
+	newBDCobj, err := c.ndmclientset.OpenebsV1alpha1().BlockDeviceClaims(bdcObj.Namespace).Update(bdcCopy)
```

Comment:
@akhilerm can we add the comment here , pointing out the behaviour we seen yesterday as well.. why we end up using `update` instead of `patch` ?

Reply:
Done @prateekpandey14